torch/nn/_functions/rnn.py

```diff
@@ -10,26 +10,29 @@
 
 
 def RNNReLUCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
-    hy = F.relu(F.linear(input, w_ih, b_ih) + F.linear(hidden, w_hh, b_hh))
+    x_h = input if w_ih is None else F.linear(input, w_ih, b_ih)
+    hy = F.relu(x_h + F.linear(hidden, w_hh, b_hh))
     return hy
 
 
 def RNNTanhCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
-    hy = F.tanh(F.linear(input, w_ih, b_ih) + F.linear(hidden, w_hh, b_hh))
+    x_h = input if w_ih is None else F.linear(input, w_ih, b_ih)
+    hy = F.tanh(x_h + F.linear(hidden, w_hh, b_hh))
     return hy
 
 
 def LSTMCell(input, hidden, w_ih, w_hh, b_ih=None, b_hh=None):
     if input.is_cuda:
-        igates = F.linear(input, w_ih)
+        igates = input.expand(4, input.size(0), input.size(1)).transpose(0, 1) if w_ih is None else F.linear(input,
```

Comment:
@SeanNaren I haven't looked in detail what's your implementation, but @justinchiu is correct, with multi-layer cudnn LSTMs only the first layer skips input, for all subsequent layers it is just regular LSTM layer. I also agree that cudnn behaviour even for the first layer should be changed, however, there is no ETA on when this could happen. 
I'm not sure multi-layer LSTMs should skip input transformation at every layer - is there a case to be made for that?
I don't have a stong opinion on the degree to which pytorch backend should be matching or not matching cudnn behaviour, and I'll defer to the opinion of core devs and power rnn users on that.